### ISSUE TITLE
COPY API rate limit error in Quickstart

### ISSUE THREAD

**Issue by andrewbt** [CartoDB/cartoframes#1362]
When running through the Quickstart Guide, I hit a COPY API rate limit error at this step: https://github.com/CartoDB/cartoframes/blob/develop/docs/developer-center/guides/04-Quickstart.md#publish-and-share-your-results

where it says to put both `to_carto` calls in the same cell:

```
stores_cdf.to_carto('starbucks_stores', if_exists='replace')
isochrones_cdf.to_carto('starbucks_isochrones', if_exists='replace')
```

I guessed that if I separated them into separate cells and waited for the first to complete before starting the next, that I would be within limits and OK. Doing this I was successful!

<img width="669" alt="Screen Shot 2019-12-13 at 7 30 28 PM" src="https://user-images.githubusercontent.com/2013555/70840263-1aa2a280-1ddf-11ea-9471-ebe1e702d43b.png">

I'd PR the docs themselves, but maybe we are looking at changing the COPY endpoint limits to better accommodate use patterns like this?

**Comment by Jesus89** [comment]
@alrocar @oleurud are we going to increase the COPY Rate limits to avoid these situations?

**Comment by alrocar** [comment]
I vote yes, please involve @cmongut in the convo

**Comment by oleurud** [comment]
We should check the current usage to define the limits. Right now, a user has at least 6 requests per minute. We should take into account this feature allow the user to upload GBs per minute to our servers

**Comment by alrocar** [comment]
I think those are two different things:

1. Revisit current rate limits (requests per second). Right now if I sign up for an Individual plan I have 3 requests per minute. That means, if I want to upload 4 small files with 10 records, CF will throw an error.

2. Add other type of rate limit: To me this is a project itself.

cc. @ana-md since they have the same situation in other project.

**Comment by cmongut** [comment]
@andrewbt could you tell us which type of account were you using?

**Comment by cmongut** [comment]
Don't worry @andrewbt, we are already checking the metrics :)

**Comment by oleurud** [comment]
Four things about this one:

### Limits
Checking our metrics I see a very low rate of requests hitting rate limits in copy (more details shared internally) So, IMO, the limits seems to be ok.

### Increasing usage
As the copy endpoints are going to be more used, we should have something to validate it easily.

### copyto
Showing your case Andrew, I see you hit rate limits 3 times. But, we should not see the error, because, in theory, we are taking that into account in CARTOframes (if CARTOframes receives a rate limit, it waits the needed time and retry the request). 

Do you see the rate limit errors?

### copyfrom
We have not implemented the retries in copyfrom and we should.

**Comment by ana-md** [comment]
Adding @VictorVelarde for visibility

**Comment by VictorVelarde** [comment]
This might affect also the COPY use in Kepler integration. Thx for keeping us up-to-date. Just a reflection we made about 'request retry': a retry is safer to get the final result, but it could add a 'slowness perception' to the user, and perhaps it would be nice to notify him (or even break) during the process because of that limit hit. That way the user could consider another options (like a better plan).

**Comment by oleurud** [comment]
Thanks @VictorVelarde, @alrocar told us just the same, so it seems we need to rethink about it. It is your call @cmongut :)

**Comment by cmongut** [comment]
We are on it. This has several inplications.

Thank you!

**Comment by cmongut** [comment]
👋 

Hi @andrewbt! I have tested the quickstart guide and I didn't hit the rate limit. It's true that I can hit it easily uploading several files.

We will reconsider the limits based on metrics and add a retry approach. We will create another issue for this.

Thank you for letting us know!

**Comment by cmongut** [comment]
Closing this and defining them here: [CartoDB/cartodb-platform/issues/5946](https://github.com/CartoDB/cartodb-platform/issues/5946)
